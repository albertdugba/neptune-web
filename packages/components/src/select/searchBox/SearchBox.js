import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import { Search as SearchIcon } from '@transferwise/icons';
import classnames from 'classnames';

const SearchBox = forwardRef(
  ({ classNames, handleSearchChange, onClick, searchPlaceholder, value }, ref) => {
    const style = (className) => classNames[className] || className;
    return (
      <li className={style('tw-dropdown-item--divider')}>
        <button
          type="button"
          className={classnames(style('tw-select-filter-link'), style('p-a-0'), 'btn-link')}
        >
          <div className={style('input-group')}>
            <span className={style('input-group-addon')}>
              <SearchIcon className={classnames(style('tw-icon'), style('tw-icon-search'))} />
            </span>
            <input
              type="text"
              className={classnames(style('tw-select-filter'), style('form-control'))}
              placeholder={searchPlaceholder}
              onChange={handleSearchChange}
              onClick={onClick}
              value={value}
              ref={ref}
              spellCheck="false"
            />
          </div>
        </button>
      </li>
    );
  },
);

SearchBox.propTypes = {
  classNames: PropTypes.objectOf(PropTypes.string),
  handleSearchChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  searchPlaceholder: PropTypes.string,
  value: PropTypes.string,
};

SearchBox.defaultProps = {
  classNames: {},
  searchPlaceholder: 'Search...',
  value: '',
};

export default SearchBox;
