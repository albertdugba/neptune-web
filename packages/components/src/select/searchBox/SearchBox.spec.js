import React from 'react';
import '@testing-library/jest-dom';

import { render } from '../../test-utils';

import SearchBox from '.';

jest.mock('react-transition-group/Transition', () => jest.fn('placeholder'));

describe('SearchBox', () => {
  const props = {
    classNames: {},
    onChange: jest.fn(),
    onClick: jest.fn(),
    placeholder: 'placeholder',
    value: 'value',
    ref: jest.fn(),
  };

  it('renders', () => {
    const { container } = render(<SearchBox {...props} />);

    expect(container).toMatchSnapshot();
  });
});
