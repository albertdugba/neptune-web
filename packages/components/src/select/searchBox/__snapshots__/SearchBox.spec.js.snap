// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders 1`] = `
<div>
  <li
    class="tw-dropdown-item--divider"
  >
    <a
      class="tw-select-filter-link p-a-0"
    >
      <div
        class="input-group"
      >
        <span
          class="input-group-addon"
        >
          <span
            aria-hidden="true"
            class="tw-icon tw-icon-search tw-icon tw-icon-search"
            role="presentation"
          >
            <svg
              fill="currentColor"
              focusable="false"
              height="16"
              width="16"
            >
              <path
                clip-rule="evenodd"
                d="M11.35 12.484a7 7 0 111.133-1.133l3.517 3.508L14.86 16l-3.51-3.515zM12.4 7A5.4 5.4 0 111.6 7a5.4 5.4 0 0110.8 0z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </span>
        <input
          class="tw-select-filter form-control"
          placeholder="placeholder"
          spellcheck="false"
          type="text"
          value="value"
        />
      </div>
    </a>
  </li>
</div>
`;
